numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -690,66 +655,56 @@ PyArray_ConvertClipmodeSequence(PyObject *object, NPY_CLIPMODE *modes, int n)
     return NPY_SUCCEED;
 }
 
+static int casting_parser(char const *str, Py_ssize_t length, void *data)
+{
+    NPY_CASTING *casting = (NPY_CASTING *)data;
+    if (length < 2) {
+        return -1;
+    }
+    switch (str[2]) {
+    case 0:
+        if (length == 2 && strcmp(str, "no") == 0) {
```

Comment:
Why the `length == 2` its not like it micro-optimizes even, since that would be the error path?

Reply:
Because otherwise we allow trailing nulls.